.golangci.yaml

```diff
@@ -1,24 +1,138 @@
-linters-settings:
-  misspell:
-    ignore-words:
-    - creater
+run:
+  timeout: 10m
+  allow-parallel-runners: true
 
 linters:
+  disable-all: true
   enable:
+  - asasalint
   - asciicheck
   - bidichk
+  - bodyclose
+  - containedctx
+  - depguard
+  - dupword
   - durationcheck
-  - errname
-  - errorlint
+  - errcheck
+  - errchkjson
   - exportloopref
+  - gocritic
+  - godot
   - gofmt
+  - goprintffuncname
+  - gosec
   - gosimple
   - govet
   - importas
   - ineffassign
   - misspell
   - nilerr
+  - noctx
+  - nolintlint
+  - nosprintfhostport
+  - prealloc
+  - revive
+  - staticcheck
+  - thelper
   - unconvert
   - unused
-  - errcheck
-  - nolintlint
+  - usestdlibvars
+  - whitespace
+  # TODO(vincepri): Figure out if we want to enable or remove the following linters:
+  # - predeclared
+  # - goconst
```

Comment:
do we have an issue for this?

Reply:
Not yet, I can open one as a follow-up and include the ones below as well

Reply:
Opened https://github.com/kcp-dev/kcp/issues/2539